blockchain/state/doc.go

```diff
@@ -18,5 +18,33 @@
 // This file is derived from core/state/statdb.go (2018/06/04).
 // Modified and improved for the klaytn development.
 
-// Package state provides a caching layer atop the Klaytn state trie.
+/*
+Package state provides a uppermost caching layer of the Klaytn state trie.
+This package is used to read/write stateObject from/to StateDB and it also acts as a bridge between
+the objects and the persistent layer.
+
+Overview of state package
+
+stateObject represents a Klaytn account identified by its address.
+Once it is loaded from the persistent layer, it is cached and managed by StateDB.
+StateDB manages stateObjects and operations to them. Most of the operations are done by StateDB,
+instead of direct access to stateObject.
+
+Source Files
+
+- database.go              : Defines Database and other interfaces used in the package
+
+- dump.go                  : Functions to dump the contents of StateDB both in raw format and indented format
+
+- journal.go               : journal and state changes to track the list of state modifications since the last state commit
+
+- state_object.go          : Implementation of stateObject
+
+- state_object_encoder.go  : stateObjectEncoder is used to encode stateObject in parallel manner
+
+- statedb.go               : Implementation of StateDB
+
+- sync.go                  : Functions to schedule a state trie download
```

Comment:
Is this for downloading from persistent layer? or is there another process to sync state tire with other nodes?

Reply:
"downloading from persistent layer" sounds a bit awkward to me. I don't think I've seen any expression "downloading from persistent layer" in any of our document. Do we have one? In Klaytn, "download" always means "getting something from other peer nodes", as far as I understood.

Reply:
Ah, okay !!! Thank you for explain !